raster_loader/io/common.py

```diff
@@ -108,6 +103,26 @@ def rasterio_metadata(
             raster_crs, "EPSG:4326", always_xy=True
         )
 
+        # assuming all windows have the same dimensions
+        a_window = next(raster_dataset.block_windows())
+        block_width = a_window[1].width
+        block_height = a_window[1].height
+        resolution = int(
```

Comment:
Since this is duplicated, we can extract a function to compute the corrected resolution and use it in the metadata and data functions

Reply:
done